chia/consensus/blockchain.py

```diff
@@ -286,17 +290,37 @@ async def receive_block(
         if fork_height is not None:
             # new coin records added
             assert coin_record_change is not None
-            return ReceiveBlockResult.NEW_PEAK, None, fork_height, coin_record_change
+            return ReceiveBlockResult.NEW_PEAK, None, fork_height, (coin_record_change, hint_changes)
         else:
-            return ReceiveBlockResult.ADDED_AS_ORPHAN, None, None, []
+            return ReceiveBlockResult.ADDED_AS_ORPHAN, None, None, ([], {})
+
+    def get_hint_list(self, npc_result: NPCResult) -> List[Tuple[bytes, bytes]]:
+        h_list = []
+        for npc in npc_result.npc_list:
+            for opcode, conditions in npc.conditions:
+                if opcode == ConditionOpcode.CREATE_COIN:
+                    for condition in conditions:
+                        if len(condition.vars) > 2 and condition.vars[2] != b"":
+                            puzzle_hash, amount_bin = condition.vars[0], condition.vars[1]
+                            amount = int_from_bytes(amount_bin)
+                            coin_id = Coin(npc.coin_name, puzzle_hash, amount).name()
+                            h_list.append((coin_id, condition.vars[2]))
```

Comment:
We should only add the memos which are less than 32 bytes right? Otherwise this is a DOS vector.
I suggest the following naming: The list of bytes in the 3rd argument should be called the memos, and the memos are stored as hints only if they are <= 32 bytes.

Reply:
clvm_rs doesn't return anything longer than 32bytes